### PR TITLE
feat: Support creating pipeline files directly into . without hooks

### PR DESCRIPTION
**Pull request by aahung**
#### Which issue(s) does this change fix?

Need to replace `APP_PIPELINE_TEMPLATES_REPO_URL` to run it.

```
APP_PIPELINE_TEMPLATES_REPO_URL = "https://github.com/aahung/aws-sam-cli-pipeline-init-templates-1.git"
```

#### Checklist

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/commands/pipeline/init/interactive_init_flow.py [hunk 1]
```diff
@@ -18,6 +19,7 @@
 from samcli.lib.cookiecutter.template import Template
 from samcli.lib.utils import osutils
 from samcli.lib.utils.git_repo import GitRepo, CloneRepoException
+from samcli.lib.utils.osutils import copytree
```
Comment (elbayaaa) [inline samcli/commands/pipeline/init/interactive_init_flow.py:22]: we have already imported `osutils` above and used it like `osutils.mkdir_temp`, can we unify the behavior? I vote for using `osutils.copytree`.

File: samcli/commands/pipeline/init/interactive_init_flow.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 """
 import logging
 import os
+import tempfile
```
Comment (elbayaaa) [inline samcli/commands/pipeline/init/interactive_init_flow.py:7]: let's use [osutils.mkdir_temp](https://github.com/aws/aws-sam-cli/blob/aa8061481bf3a5238f0047fd6978702111a9ac36/samcli/lib/utils/osutils.py#L17)

File: samcli/commands/pipeline/init/interactive_init_flow.py [hunk 3]
```diff
@@ -107,7 +109,23 @@ def _generate_from_pipeline_template(pipeline_template_dir: Path) -> None:
     pipeline_template: Template = _initialize_pipeline_template(pipeline_template_dir)
     bootstrap_context: Dict = _load_pipeline_bootstrap_context()
     context: Dict = pipeline_template.run_interactive_flows(bootstrap_context)
-    pipeline_template.generate_project(context)
+    with tempfile.TemporaryDirectory() as tempdir:
+        LOG.debug("Generating pipeline files into %s", tempdir)
+        context["outputDir"] = "."  # prevent cookiecutter from generating a sub-folder
```
Comment (elbayaaa) [inline samcli/commands/pipeline/init/interactive_init_flow.py:114]: we always set the "outputDir" to "." https://github.com/aws/aws-sam-cli/blob/ae4b8b64fbeb5c87611d64eb0786a702a224fdf7/samcli/lib/cookiecutter/template.py#L149

File: samcli/commands/pipeline/init/interactive_init_flow.py [hunk 4]
```diff
@@ -107,7 +109,23 @@ def _generate_from_pipeline_template(pipeline_template_dir: Path) -> None:
...
+        pipeline_template.generate_project(context, tempdir)
+        _copy_dir_contents_fail_on_exist(tempdir, ".")
```
Comment (elbayaaa) [inline samcli/commands/pipeline/init/interactive_init_flow.py:116]: better to use ` os.getcwd()` ?

File: samcli/commands/pipeline/init/interactive_init_flow.py (hunk 3 above)
Comment (aahung) [reply samcli/commands/pipeline/init/interactive_init_flow.py:113]: these are two different things. cookiecutter has an `output_dir` and then our template has an `outputDir`.

The former is where the latter is created in.

File: samcli/commands/pipeline/init/interactive_init_flow.py (hunk 2 above)
Comment (aahung) [reply samcli/commands/pipeline/init/interactive_init_flow.py]: changed

File: samcli/commands/pipeline/init/interactive_init_flow.py (hunk 4 above)
Comment (aahung) [reply samcli/commands/pipeline/init/interactive_init_flow.py]: No idea whether either of which is better. I changed the method name and use cwd

Comment (aahung) [reply samcli/commands/pipeline/init/interactive_init_flow.py]: changing back to "."

I added the feature to print out the list of generated files. `os.getcwd` will print out absolute paths and it looks verbose on a list of file paths.

Comment (aahung) [reply samcli/commands/pipeline/init/interactive_init_flow.py]: image

**Review by hoffa — APPROVED** [review]

**Review by elbayaaa — APPROVED** [review]

**Pull request merged**